src/globemap.ts

```diff
@@ -472,33 +518,147 @@ module powerbi.extensibility.visual {
+                            let levelTiles = this.generateQuadsByLevel(level, metadata.imageUrl, metadata.imageUrlSubdomains);
+                            this.mapTextures.push(this.createTexture(level, levelTiles));
+                            tileCache.push(levelTiles);
+                        }
+                        this.LocalStorageService.setData(GlobeMap.TILE_STORAGE_KEY, tileCache);
+                        this.LocalStorageService.setData(GlobeMap.TILE_LANGUAGE_CULTURE, this.currentLanguage);
+                        return tileCache;
+                    });
+            } else {
+                for (let level: number = 1; level <= GlobeMap.maxResolutionLevel; ++level) {
+                    this.mapTextures.push(this.createTexture(level, tileCache[level - 1]));
                 }
+                return jQuery.when(tileCache);
             }
+        }
 
-            // Can't execute in for loop because variable assignement gets overwritten
-            const createTexture: (canvas: JQuery) => THREE.Texture = (canvas: JQuery) => {
-                const texture: THREE.Texture = new THREE.Texture(<HTMLCanvasElement>canvas.get(0));
-                texture.needsUpdate = true;
-                canvas.on("ready", () => {
-                    texture.needsUpdate = true;
-                    this.needsRender = true;
+        private getBingMapsServerMetadata(): JQueryPromise<BingResourceMetadata> {
+            return $.ajax(GlobeMap.metadataUrl)
+                .then((data: BingMetadata) => {
+                    if (data.resourceSets.length) {
+                        let resourceSet = data.resourceSets[0];
+                        if (resourceSet && resourceSet.resources.length) {
+                            return resourceSet.resources[0];
+                        }
+                    }
+                    throw "Bing Maps API response was changed. Please update code for new version";
+                })
+                .fail((error: any) => {
+                    console.error(JSON.stringify(error));
+                    return GlobeMap.reserveBindMapsMetadata;
                 });
-                return texture;
+        }
 
+        /**
+         * Generate Bing tile object by map level
+         * @see https://msdn.microsoft.com/en-us/library/bb259689.aspx
+         * @private
+         * @param {number} level map lavel
+         * @param {string} urlTemplate url template
+         * @example https://ecn.{subdomain}.tiles.virtualearth.net/tiles/r{quadkey}.jpeg?g=5691&mkt={culture}&shading=hill
+         * @param {string[]} subdomains list of subdomauns
+         * @returns {{ [quadKey: string]: string }} Object <quadKey> : <image url>
+         * @memberOf GlobeMap
+         */
+        private generateQuadsByLevel(level: number, urlTemplate: string, subdomains: string[]): TileMap {
+            const COUNT_SERVERS = subdomains.length;
```

Comment:
Please specify a type of the variable.
It's a local variable, I suppose that it should be in pascal case.